src/services/authentication/file/file-based-authentication.ts

```diff
@@ -34,6 +34,9 @@ export class FileBasedAuthentication extends DeepstreamPlugin implements Deepstr
     super()
     this.validateSettings(settings)
     this.base64KeyLength = 4 * Math.ceil(this.settings.keyLength / 3)
+    if (this.settings.reportInvalidParameters === undefined) {
```

Comment:
should it not be the other way round to stop it from being a breaking change? Normal behaviour is to chain them right?

Reply:
Before this pull:
- the only behaviour was to return `{isValid: false}` thus there was no option to use more than one auth provider since it returned invalid on the first try always.

After this pull:
- To maintain current behaviour, if no reportInvalidParameters key is defined and there is only one auth strategy, it will be set to true in order to return `{isValid: false}`.

- And to enable the new behaviour, on the config initialiser we check if there is more than one auth strategy, and if the reportInvalidParameters option is not true, we set it to false in order to allow for multiple auth strategies.

Does it make sense ? Sometimes I got confused on which behaviour to keep or change...